middlewares/redirect/redirect.go

```diff
@@ -85,6 +86,28 @@ func (h *handler) ServeHTTP(rw http.ResponseWriter, req *http.Request, next http
 		return
 	}
 
+	if prefix, prefixOk := req.Context().Value(middlewares.StripPrefixKey).(string); prefixOk {
```

Comment:
You should maybe simplify this part like that

```go
if stripPrefix, stripPrefixOk := req.Context().Value(middlewares.StripPrefixKey).(string); stripPrefixOk {
	if len(stripPrefix) > 0 {
		tempPath := parsedURL.Path
		parsedURL.Path = stripPrefix
		if len(tempPath) > 0 && tempPath != "/" {
			parsedURL.Path = stripPrefix + tempPath
		}

		if trailingSlash, trailingSlashOk := req.Context().Value(middlewares.StripPrefixSlashKey).(bool); trailingSlashOk {
			if trailingSlash {
				if !strings.HasSuffix(parsedURL.Path, "/") {
					parsedURL.Path = fmt.Sprintf("%s/", parsedURL.Path)
				}
			}
		}
	}
}

if addPrefix, addPrefixOk := req.Context().Value(middlewares.AddPrefixKey).(string); addPrefixOk {
	if len(addPrefix) > 0 {
		parsedURL.Path = strings.Replace(parsedURL.Path, addPrefix, "", 1)
	}
}
```

You update `parsedURL` instead of re parsing the URL multiple times
WDYT?